vpr/src/place/timing_place_lookup.cpp

```diff
@@ -505,6 +669,25 @@ static vtr::Matrix<float> compute_delta_delays(
     }
     VTR_ASSERT(src_type != nullptr);
 
+    std::function<void(
+        RouterDelayProfiler&,
+        vtr::Matrix<std::vector<float>>&,
+        int,
+        int,
+        int,
+        int,
+        int,
+        int,
+        const t_router_opts&,
+        bool,
+        const std::set<std::string>&)>
+        generic_compute_matrix;
```

Comment:
We don't use std::function very much (or at all?) in VPR. I think this should have a comment outlining what it is for and what its doing.

I think its so you can alias generic_compute_matrix to the relevant underlying function.  It might be clearer if we created a typedef (e.g. t_generic_compute_matrix_func).

Any notes on why you chose this approach (e.g. vs defining a generic compute matrix class)?

Reply:
`defining a generic compute matrix class` - You mean an interface class?  That seems overkill for an interface with 1 method and little state?

Reply:
> We don't use std::function very much (or at all?) in VPR

Sorry,  are you requesting a comment to explain what std::function is doing?  I want to be clear.

Reply:
After reflecting on this more, I'm leaning towards just typedef'ing the std::function at the top of the file, and instantiating it here. Thoughts?

Reply:
Done